src/main/java/com/intershop/oms/test/servicehandler/omsdb/v1/OMSDbHandlerV1.java

```diff
@@ -222,12 +222,55 @@ public Long runDBStmtLongById(String query, Long id, String resultColumnName)
     }
 
     @Override
-    public ArrayList<String> runDBStmtString(String query, String resultColumnName)
+    public List<Long> runDBStmtLongListById(String query, Long id, String resultColumnName)
     {
-        ArrayList<String> result = new ArrayList<>();
+        List<Long> result = new ArrayList<>();
+        ResultSet resultSet = null;
+        try (Connection connection = getConnection();
+             PreparedStatement sqlStatement = connection.prepareStatement(query))
+        {
+            if (null != id)
+            {
+                sqlStatement.setLong(1, id);
+            }
+
+            log.info("Calling '" + sqlStatement.toString() + "'.");
+            resultSet = sqlStatement.executeQuery();
+
+            while (resultSet.next())
+            {
+                result.add(resultSet.getLong(resultColumnName));
+            }
+        }
+        catch (SQLException sqlEx)
+        {
+            log.error("SQLException getting integer result '" + resultColumnName + "' from '" + query + "':" + sqlEx.getMessage());
+            throw new RuntimeException(sqlEx);
+        }
+        finally
+        {
+            if (null != resultSet)
+            {
+                try
+                {
+                    resultSet.close();
+                }
+                catch(SQLException e) { }
+            }
+        }
+
+        log.info("Got {} Long results for column '{}' from '{}' using id '{}'.", result.size(), resultColumnName, query, id);
+
+        return result;
+    }
+
+    @Override
+    public List<String> runDBStmtString(String query, String resultColumnName)
```

Comment:
why not just call    runDBStmtLongListById(query, null::Long , resultColumnName)

Reply:
minimal invasive - ArrayList is just not a return type...